tahoe_idp/magiclink_views.py

```diff
@@ -34,4 +38,19 @@ def get(self, request, *args, **kwargs):
     def login_complete_action(self) -> HttpResponseRedirect:
         token = self.request.GET.get('token')
         magiclink = MagicLink.objects.get(token=token)
-        return HttpResponseRedirect(magiclink.redirect_url or settings.LOGIN_REDIRECT_URL)
+        return HttpResponseRedirect(magiclink.redirect_url or django_settings.LOGIN_REDIRECT_URL)
+
+
+class StudioLoginAPIView(View):
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
+        return super(StudioLoginAPIView, self).dispatch(request, *args, **kwargs)
+
+    def get(self, request):
+        username = request.user.username
+        next_url = request.GET.get('next') or '/home'
```

Comment:
No need for hardcoded `/home` here. 

Also the redirect url should be validated for security.

```suggestion
        next_url = request.GET.get('next')
        if next_url and not is_valid_redirect_url(next_url):
            next_url = None
```

Reply:
what kind of checks should we add to that helper? @OmarIthawi

Reply:
@shadinaif please use a modified copy of the `is_safe_login_or_logout_redirect` helper function from Open edX:

 - https://github.com/appsembler/edx-platform/blob/234eedd8c6d972471ebf8c844012d9fc844eacb9/openedx/core/djangoapps/user_authn/utils.py#L15-L46

The function is a very helpful function to mitigate against phishing attacks and password hijacking. We don't want our system to redirect to `redirect_url=https://attacker.com`. All redirects should be in a specific whitelist.

Reply:
Addressed in https://github.com/appsembler/tahoe-idp/pull/55